src/ctl.c

```diff
@@ -2983,6 +2985,55 @@ arena_i_retain_grow_limit_ctl(tsd_t *tsd, const size_t *mib,
 	return ret;
 }
 
+static int
+arena_i_name_ctl(tsd_t *tsd, const size_t *mib, size_t miblen,
+    void *oldp, size_t *oldlenp, void *newp, size_t newlen) {
+	int ret;
+	unsigned arena_ind;
+	char *name = NULL;
+
+	READ_XOR_WRITE();
```

Comment:
This check may do `goto label_return;` so it needs to be covered by the lock below.

Reply:
Moved below the lock